pkg/server/plugin/notifier/k8sbundle/k8sbundle.go

```diff
@@ -139,6 +144,22 @@ func (p *Plugin) Configure(ctx context.Context, req *spi.ConfigureRequest) (resp
 	}
 
 	p.setConfig(config)
+
+	// Start watcher to set CA Bundle in objects created after server has started
+	watcher := newBundleWatcher(p)
+	go func() {
+		err := watcher.Start()
```

Comment:
The bundle watcher usage pattern is a little weird. As you've pointed out there is nothing you can do if start fails but log, but that is because the initialization of the watchers and the actual watching is all lumped under start. What if newBundleWatcher did the initialization of the watchers  and returned an error on failure? Like so?
```
watcher, err := newBundleWatcher(p)
if err != nil {
    return nil, err
}
var wg sync.WaitGroup
ctx, cancel := context.WithBackground(context.Background()
wg.Add(1)
go func() {
    defer wg.Done()
    if err := watcher.Watch(ctx); err != nil && !error.Is(err, context.Canceled) {
        p.log.Error("Unable to watch: %v", err)
    |
}()

p.mu.Lock()
if p.cancelWatcher != nil {
    p.cancelWatcher()
}
p.cancelWatcher = func() {
    cancel()
    wg.Wait()
}
p.mu.Unlock()
```

Also, YMMV, but I've found the Start()/Stop() pattern  harder to test than a synchronous `Watch(ctx context.Context)` method.

Reply:
Thanks for the suggestion, I wasn't happy with the start/stop pattern either.

Reply:
This change is in